### PR TITLE
fix: auto-subdivide CloudWatch chunks that exceed 10k record limit

### PR DESCRIPTION
**Pull request by decko**
## Summary

When a 5-minute CloudWatch Logs Insights query chunk returns exactly 10,000 records (the API maximum), the exporter now automatically bisects the time window and retries each half. This prevents silent data loss during high-traffic periods.

## Changes

- Extracted `_execute_query()` helper for the start-query/poll loop
- Added `_fetch_chunk()` with recursive bisection on truncation detection
- Depth limit of 5 prevents infinite loops (~9s minimum window)
- `fetch_cloudwatch_logs()` signature unchanged — callers unaffected
- 6 new unit tests covering subdivision, recursion, depth limit, and midpoint math

## Testing

- 45/45 tests pass (6 new + 39 existing)
- Validated against prod CloudWatch data: 4 truncation events in a 3-hour window, all resolved
- ~18k records recovered that would have been silently dropped
- Two-level subdivision observed and working correctly

## Jira

Closes: PULP-1887

---
Assisted-by: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

## Summary by Sourcery

Handle CloudWatch Logs Insights result truncation by subdividing large time windows while preserving the existing public API.

Bug Fixes:
- Prevent silent data loss when a 5-minute CloudWatch Logs Insights chunk hits the 10K record limit by automatically subdividing the time window and re-querying.

Enhancements:
- Extract a reusable helper for executing and polling CloudWatch Logs Insights queries.
- Introduce a recursive chunk-fetching helper that bisects time ranges on truncation with a configurable maximum subdivision depth.
- Refine CloudWatch log query construction and result conversion code style for consistency.

Tests:
- Add unit tests covering chunk subdivision behavior, recursive truncation handling, maximum subdivision depth, midpoint calculation, and multi-chunk integration with the public fetch function.

### CODE REVIEW

File: management_tools/pulp-access-logs-exporter/src/pulp_access_logs_exporter/cloudwatch.py [hunk 1]
```diff
@@ -45,22 +47,124 @@ def build_query(filter_paths: str = "/api/pypi/", exclude_paths: str = "/livez,/
...
+    print(
+        f"{'  ' * depth}Querying logs from {chunk_start.isoformat()} to {chunk_end.isoformat()}..."
+    )
+
+    result = _execute_query(
+        logs_client,
+        log_group,
+        query,
+        int(chunk_start.timestamp()),
+        int(chunk_end.timestamp()),
```
[the remark this answers is not in the document]
Comment (decko) [reply management_tools/pulp-access-logs-exporter/src/pulp_access_logs_exporter/cloudwatch.py:132]: The entire exporter uses `print()` — cli.py has 30+ print statements, content_cloudwatch.py uses `print(..., file=sys.stderr)`. Migrating to `logging` is a valid cleanup but it's a whole-file refactor across all modules, not something to mix into a bug fix.

Regarding `statistics.recordsMatched` as truncation signal: `recordsMatched` reports how many records matched the query filters, not how many were returned. CloudWatch can match 50k but only return 10k. The returned count hitting the API limit is the actual truncation signal — that's what AWS documents. `recordsMatched` can be 0 or absent depending on query state, making it less reliable as primary signal.
